source/features/clean-image-auto-links.tsx

```diff
@@ -0,0 +1,32 @@
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import {isSingleMarkdownFile} from '../github-helpers';
+import GitHubURL from '../github-helpers/github-url';
+
+function init(): void {
+	for (const image of select.all('.markdown-body a > img')) {
+		const link = image.parentElement as HTMLAnchorElement;
+
+		if (image.src === link.href) {
+			link.replaceWith(image);
+		}
+
+		if (pageDetect.isSingleFile(link)) {
+			const rawUrl = new GitHubURL(link.href).assign({route: 'raw'});
```

Comment:
I’m conflicted about this. The link in this case is not the same and it makes it easier to edit the linked local file

Reply:
I actually thought this is included in the original requests.

Reply:
Indeed, there are very few situations where this would be useful (e.g. hand-written SVGs)

Reply:
If you turn this line into a function, you can avoid having another condition below. Example:

if isSingleFile(link) && image.src === getRawUrl(link)
  link.remove()